### PR TITLE
Add React.PureComponent to the list of default super classes

### PR DESCRIPTION
**Pull request by rafeca**
This PR enables the HMR transform to work on components that extend `React.PureComponent`

### CODE REVIEW

File: test/fixtures/code-exports/actual.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
-export default class Foo extends React.Component {}
-export default class extends React.Component {}
```
Comment (gaearon) [inline test/fixtures/code-exports/actual.js]: What is this change related to?

**Comment by gaearon** [comment]
Please ping me in person or via messenger next time :-) I don't actively watch this repository.

File: test/fixtures/code-exports/actual.js (hunk 1 above)
Comment (rafeca) [reply test/fixtures/code-exports/actual.js]: By the current spec only one default export can exist per module (I guess this has changed since this package was created). Babel now complains about this (maybe because the dependencies in this repo are not locked since it does not use yarn).

**Comment by gaearon** [comment]
3.0.0
